python/pyspark/mllib/linalg/__init__.py

```diff
@@ -1296,9 +1296,19 @@ def asML(self):
         return newlinalg.SparseMatrix(self.numRows, self.numCols, self.colPtrs, self.rowIndices,
                                       self.values, self.isTransposed)
 
-    # TODO: More efficient implementation:
     def __eq__(self, other):
-        return np.all(self.toArray() == other.toArray())
+        if ((type(other) is not type(self)) or
+            self.isTransposed != other.isTransposed or
```

Comment:
So what about if one matrix is represented as transposed and the other isn't but the both represent the same actual data?